### PR TITLE
Refactor to use shared flume-base

### PR DESCRIPTION
**Pull request by johnsca**

### CODE REVIEW

File: resources.yaml [hunk 1]
```diff
@@ -1,11 +0,0 @@
-options:
-  output_dir: /home/ubuntu/resources
-optional_resources:
-  flume-x86_64:
-    url: https://s3.amazonaws.com/jujubigdata/apache/x86_64/apache-flume-1.6.0-bin-defd21a.tar.gz
-    hash: defd21ad8d2b6f28cc0a16b96f652099
-    hash_type: md5
-  zookeeper-x86_64:
```
Comment (ktsakalozos) [inline resources.yaml:8]: Zookeeper is needed or else we hit this exception:

```
01 Mar 2016 16:19:44,961 ERROR [lifecycleSupervisor-1-1] (org.apache.flume.lifecycle.LifecycleSupervisor$MonitorRunnable.run:253)  - Unable to start PollableSourceRunner: { s
ource:org.apache.flume.source.kafka.KafkaSource{name:r1,state:IDLE} counterGroup:{ name:null counters:{} } } - Exception follows.
java.lang.NoClassDefFoundError: org/apache/zookeeper/Watcher
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
        at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
        at kafka.consumer.ZookeeperConsumerConnector.connectZk(ZookeeperConsumerConnector.scala:156)
        at kafka.consumer.ZookeeperConsumerConnector.<init>(ZookeeperConsumerConnector.scala:114)
        at kafka.javaapi.consumer.ZookeeperConsumerConnector.<init>(ZookeeperConsumerConnector.scala:65)
        at kafka.javaapi.consumer.ZookeeperConsumerConnector.<init>(ZookeeperConsumerConnector.scala:67)
        at kafka.consumer.Consumer$.createJavaConsumerConnector(ConsumerConnector.scala:100)
        at kafka.consumer.Consumer.createJavaConsumerConnector(ConsumerConnector.scala)
        at org.apache.flume.source.kafka.KafkaSourceUtil.getConsumer(KafkaSourceUtil.java:47)
        at org.apache.flume.source.kafka.KafkaSource.start(KafkaSource.java:200)
        at org.apache.flume.source.PollableSourceRunner.start(PollableSourceRunner.java:74)
        at org.apache.flume.lifecycle.LifecycleSupervisor$MonitorRunnable.run(LifecycleSupervisor.java:251)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:304)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:178)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: org.apache.zookeeper.Watcher
        at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
        ... 29 more
```

File: lib/charms/flume.py [hunk 2]
```diff
@@ -1,143 +0,0 @@
...
-        
-        :param bool force: Force the installation execution even if this is not the first installation attempt.
-        '''
-        if not force and self.is_installed():
-            return
-        jujuresources.install(self.resources['flume'],
-                              destination=self.dist_config.path('flume'),
-                              skip_top_level=True)
-        # FlumeSource needs the zookeeper jars
-        jujuresources.install(self.resources['zookeeper'],
```
Comment (ktsakalozos) [inline lib/charms/flume.py:40]: Zookeeper is installed here.

File: lib/charms/flume.py [hunk 3]
```diff
@@ -1,143 +0,0 @@
...
-        default_conf.copytree(flume_conf)
-        # Now remove the conf included in the tarball and symlink our real conf
-        default_conf.rmtree_p()
-        flume_conf.symlink(default_conf)
-
-        flume_env = self.dist_config.path('flume_conf') / 'flume-env.sh'
-        if not flume_env.exists():
-            (self.dist_config.path('flume_conf') / 'flume-env.sh.template').copy(flume_env)
-        utils.re_edit_in_place(flume_env, {
-            r'.*FLUME_CLASSPATH.*': 'FLUME_CLASSPATH={}/*'.format(self.dist_config.path('zookeeper')),
```
Comment (ktsakalozos) [inline lib/charms/flume.py:67]: And here we inject the jars of zk to the classpath

Comment (johnsca) [inline lib/charms/flume.py:67]: Added this to the base in juju-solutions/layer-apache-flume-base#4 and the call to `configure_zookeeper` in this PR (that I missed before this comment)

**Comment by ktsakalozos** [comment]
Tested, looks fine now

**Pull request merged**